### PR TITLE
test(web-audio-player): add tests

### PR DESCRIPTION
**Pull request by Lehren**
Add tests to raise line coverage to above the minimum 5 percent.

### CODE REVIEW

File: test/web-audio-playerSpec.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+const WebAudioPlayer = require('../web-audio-player');
+
+describe('WebAudioPlayer', function() {
+  it('should call the right event listeners', function() {
```
Comment (remcohaszing) [inline test/web-audio-playerSpec.js]: I think "correct" seems more appropriate than "right".

File: test/web-audio-playerSpec.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    var errorCodes = [0, 1, 2, 3, 4];
+    var audioOriginal = window.Audio;
+    var audioMock =
```
Comment (remcohaszing) [inline test/web-audio-playerSpec.js]: Just use spies, so no manual cleanup is needed.

```js
spyOn(window, 'Audio').and.returnValue(audioMock);
```

File: test/web-audio-playerSpec.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    var options = {
+      playingCb: jasmine.createSpy('callback'),
+      timeupdateCb: jasmine.createSpy('callback'),
+      durationchangeCb: jasmine.createSpy('callback'),
+      canplayCb: jasmine.createSpy('callback'),
+      endedCb: jasmine.createSpy('callback'),
+      pauseCb: jasmine.createSpy('callback'),
+      progressCb: jasmine.createSpy('callback'),
+      errorCb: jasmine.createSpy('callback')
+    };
```
Comment (remcohaszing) [inline test/web-audio-playerSpec.js]: The options can be simplified by using:

```js
let options = jasmine.createSpyObj('options', [
  'playingCb',
  'timeupdateCb',
  // etc...
])
```

File: test/web-audio-playerSpec.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    expect(options.canplayCb)
+      .toHaveBeenCalledTimes(1);
+    expect(options.endedCb)
+      .toHaveBeenCalledTimes(1);
+    expect(options.pauseCb)
+      .toHaveBeenCalledTimes(1);
+    expect(options.progressCb)
+      .toHaveBeenCalledTimes(1);
+    expect(options.errorCb)
+      .toHaveBeenCalledTimes(5);
```
Comment (remcohaszing) [inline test/web-audio-playerSpec.js]: Why have these lines been broken?

File: test/web-audio-playerSpec.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+    expect(console.error).toHaveBeenCalledTimes(5);
+    expect(console.error).toHaveBeenCalledWith('You aborted the playback.');
+    expect(console.error).toHaveBeenCalledWith('A network error caused the audio download to fail.');
+    expect(console.error).toHaveBeenCalledWith('The audio playback was aborted due to a corruption ' +
+      'problem or because the media used features your ' +
+      'browser did not support.');
```
Comment (remcohaszing) [inline test/web-audio-playerSpec.js]: No need to break this string over 3 lines. 2 will suffice.

File: test/web-audio-playerSpec.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    expect(console.error).toHaveBeenCalledWith('The audio could not be loaded, either because the ' +
+      'server or network failed or because the format is ' +
+      'not supported.');
```
Comment (remcohaszing) [inline test/web-audio-playerSpec.js]: No need to break this string over 3 lines. 2 will suffice.

**Comment by Lehren** [comment]
Fixed in latest commit
